pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -476,6 +577,31 @@ func (bfs *Service) readHeader(rc io.ReadCloser) (any, error) {
 	return &b.Header, r.Err
 }
 
+func (bfs *Service) readState(rc io.ReadCloser) (any, error) {
+	r := gio.NewBinReaderFromIO(rc)
+	_ = r.ReadB()
+	_ = r.ReadU32LE()
+	_ = r.ReadU32LE()
+	var stateRoot util.Uint256
+	r.ReadBytes(stateRoot[:])
```

Comment:
These values may be checked against the expected StateSyncPoint. It's just a sanity-check, but we'd better be sure that state file for the proper height is retrieved.